x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageDeciderService.java

```diff
@@ -722,6 +798,8 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.field("reason", reason);
             builder.field("unassigned", unassigned);
             builder.field("assigned", assigned);
+            builder.field("unassignedShardIds", unassignedShardIds);
+            builder.field("assignedShardIds", assignedShardIds);
```

Comment:
I wonder if we want to show all the assigned shards that might get large and I think it doesn't provide much information about the autoscaling decision.

Reply:
Hm, I don't have a strong opinion on that. I think it makes sense to have this field for the consistency sake, but I can also see the side of the argument of avoid bloat in the autoscaling decision response.

Reply:
It is not really all the assigned shards. It is only those that cannot move but needs to move AND where either the reason they cannot remain is only storage OR the reason they cannot move somewhere else is storage. So it is much like the unassigned list and should not grow large except if the cluster has issues.

We should still limit the list we show though to the first X shards.